web/src/main/java/org/springframework/security/web/firewall/StrictHttpFirewall.java

```diff
@@ -352,6 +367,13 @@ private void rejectedBlacklistedUrls(HttpServletRequest request) {
 		}
 	}
 
+	private void rejectedUntrustedHosts(HttpServletRequest request) {
+		String serverName = request.getServerName();
+		if (this.allowedHostnames != null && !this.allowedHostnames.test(serverName)) {
```

Comment:
Sorry, the concern isn't that `this.allowedHostnames` would be `null` at this point, but that `serverName` would be `null`. Because of the change to `DummyRequest`, it's possible for `getServerName` to return `null`.

So, we'd want to do:

```suggestion
		if (serverName != null && !this.allowedHostnames.test(serverName)) {
```